packages/test/src/admin-utils.ts

```diff
@@ -192,8 +224,144 @@ export async function createCollectionContainingProduct(
+  adminClient: SimpleGraphQLClient,
+  product: Product,
+  optionGroups: CreateProductOptionGroupInput[]
+): Promise<Product> {
+  const createdOptionGroups: any[] = await Promise.all(
+    optionGroups.map(async (g) => {
+      return (
+        (await adminClient.query<
+          ProductOptionGroup,
+          MutationCreateProductOptionGroupArgs
+        >(CREATE_PRODUCT_OPTION_GROUP, {
+          input: {
+            code: g.code,
+            options: g.options,
+            translations: [
+              {
+                languageCode: LanguageCode.En,
+                name: g.translations[0].name,
+              },
+            ],
+          },
+        })) as any
+      ).createProductOptionGroup;
+    })
+  );
+  // for(const optionIndex in createdOptionGroups){
+  //   createdOptionGroups[optionIndex].options=[...(
+  //     await Promise.all(await optionGroups.map(async(o)=>{
+  //       return (await adminClient.query<ProductOption, MutationCreateProductOptionArgs>(
+  //       CREATE_PRODUCT_OPTION,
+  //       { input:{
+  //         productOptionGroupId: createdOptionGroups[optionIndex].createProductOptionGroup.id,
+  //         code: o.code,
+  //         translations:[
+  //           {
+  //             languageCode: LanguageCode.En,
+  //             name: o.translations[0].name
+  //           }
+  //         ]
+  //       } }
+  //     ) as any).createProductOption}))
+  //   )]
+  // }
+  let newProduct = product;
+  for (const optionGroup of createdOptionGroups) {
+    newProduct = (
+      (await adminClient.query<Product, MutationAddOptionGroupToProductArgs>(
+        ADD_OPTION_GROUP_TO_PRODUCT,
+        {
+          optionGroupId: optionGroup.id as string,
+          productId: newProduct.id,
+        }
+      )) as any
+    ).addOptionGroupToProduct;
+  }
+  return newProduct;
+}
+
+export async function createProductVariants(
+  adminClient: SimpleGraphQLClient,
+  product: Product
+): Promise<ProductVariant[]> {
+  const crossJoinedOptions = cartesian(
+    product.optionGroups[0].options,
+    product.optionGroups[1].options
   );
+  const variants = await adminClient.query<
+    ProductVariant[],
+    MutationCreateProductVariantsArgs
+  >(CREATE_PRODUCT_VARIANTS, {
+    input: crossJoinedOptions.map((i: ProductOption[]) => {
+      return {
+        // optionGroups: product.optionGroups,
+        productId: product.id,
+        price: 10000,
+        stockOnHand: 100,
+        sku: i.map((o) => o.code).join(' '),
+        trackInventory: GlobalFlag.True,
+        optionIds: i.map((o) => o.id),
+        translations: [
+          {
+            languageCode: LanguageCode.En,
+            name: `${product.translations[0].name} ${i
+              .map((o) => o.code)
+              .join(' ')}`,
+          },
+        ],
+      };
+    }),
+  });
+  return (variants as any).createProductVariants;
+}
+
+function cartesian(one: ProductOption[], two: ProductOption[]) {
```

Comment:
It does a cartesian product of two sets(or js arrays in our case) to get ProductOption arrays for each variants by combining option values of ProductOptionGroups of a product. I will rename it to avoid confusion I guess